### PR TITLE
[SELC-5483] Added new WorkflowExecutorIncrementRegistrationAggregator

### PR DESCRIPTION
**Pull request by flaminiaScarciofolo**
#### List of Changes
- Added new WorkflowExecutorIncrementRegistrationAggregator

#### Motivation and Context
We need to manage aggregates increment using new workflow to verify if aggregate already exists before onboarding them

#### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: apps/onboarding-functions/src/test/java/it/pagopa/selfcare/onboarding/functions/OnboardingFunctionsTest.java [hunk 1]
```diff
@@ -201,6 +198,69 @@ void onboardingOrchestratorContractRegistrationAggregator_Pending(){
...
 
+    @Test
+    void onboardingOrchestratorIncrementRegistrationAggregator_Pending_delgationAlreadyExists(){
+        Onboarding onboarding = new Onboarding();
+        onboarding.setId("onboardingId");
+        onboarding.setStatus(OnboardingStatus.PENDING);
+        AggregateInstitution aggregate1 = new AggregateInstitution();
+        aggregate1.setTaxCode("code1");
+        AggregateInstitution aggregate2 = new AggregateInstitution();
+        aggregate1.setTaxCode("code2");
```
Comment (gunzip) [inline apps/onboarding-functions/src/test/java/it/pagopa/selfcare/onboarding/functions/OnboardingFunctionsTest.java:209]:
```suggestion
        aggregate2.setTaxCode("code2");
```

File: apps/onboarding-functions/src/test/java/it/pagopa/selfcare/onboarding/functions/OnboardingFunctionsTest.java [hunk 2]
```diff
@@ -201,6 +198,69 @@ void onboardingOrchestratorContractRegistrationAggregator_Pending(){
...
+
+    @Test
+    void onboardingOrchestratorIncrementRegistrationAggregator_Pending_delgationNotExists(){
+        Onboarding onboarding = new Onboarding();
+        onboarding.setId("onboardingId");
+        onboarding.setStatus(OnboardingStatus.PENDING);
+        AggregateInstitution aggregate1 = new AggregateInstitution();
+        aggregate1.setTaxCode("code1");
+        AggregateInstitution aggregate2 = new AggregateInstitution();
+        aggregate1.setTaxCode("code2");
```
Comment (gunzip) [inline apps/onboarding-functions/src/test/java/it/pagopa/selfcare/onboarding/functions/OnboardingFunctionsTest.java:237]:
```suggestion
        aggregate2.setTaxCode("code2");
```

File: apps/onboarding-functions/src/main/java/it/pagopa/selfcare/onboarding/service/CompletionServiceDefault.java [hunk 3]
```diff
@@ -236,6 +238,20 @@ public void sendTestEmail(ExecutionContext context) {
         notificationService.sendTestEmail(context);
     }
 
+    @Override
+    public String existsDelegation(OnboardingAggregateOrchestratorInput input) {
+        boolean existsDelegation = false;
+
+        if (Objects.nonNull(input) && Objects.nonNull(input.getInstitution()) && Objects.nonNull(input.getAggregate())) {
+                DelegationWithPaginationResponse delegationWithPaginationResponse = delegationApi.getDelegationsUsingGET1(null, input.getInstitution().getId(), null, null,
```
Comment (gunzip) [inline apps/onboarding-functions/src/main/java/it/pagopa/selfcare/onboarding/service/CompletionServiceDefault.java:246]: `getDelegationsUsingGET1` may throw, do we have to catch any exception here?

**Comment by andrea-putzu** [comment]
Why add another workflow? Area riservata should check every time if institution already onboarded

**Comment by manuraf** [comment]
`INCREMENT_REGISTRATION_AGGREGATOR`, the new workflow, is different about classic Aggregator one...It skips some functions that are not needed in case of aggregation increment (_CreateInstitution_, _CreateOnboarding_ and _CreateUsers_). In addition, it checks if delegations already exists and then send different mails...we have explained that [here](https://pagopa.atlassian.net/wiki/spaces/SCP/pages/867827998/Design+Review+-+Enti+Aggregatori#Caricamento-incrementale-degli-aggregati-tramite-CSV.1)

**Comment by andrea-putzu** [comment]
I believe that a check for the existence and validity of an onboarding process can be included in the workflow, which would make the code identical.

**Comment by manuraf** [comment]
IMHO sometimes could be more clear for developers to have some lines of code in addition rather than to have a lot of IF statements

**Comment by andrea-putzu** [comment]
It seems that doing this approach will have two workflow: one responsible for activities A and B and another only for B. Wouldn't it be better to refactor and have a possible dependency between the first and the second?

**Comment by manuraf** [comment]
we could move activity B into a shared method such as _createInstitutionAndOnboarding_ o _onboardingCompletionUsersActivity_

**Pull request merged**